internal/anchorplatform/platform_api_service.go

```diff
@@ -103,6 +111,76 @@ func (a *AnchorPlatformAPIService) UpdateAnchorTransactions(ctx context.Context,
 	return nil
 }
 
+type GetTransactionsQueryParams struct {
+	SEP        string   `schema:"sep,required,omitempty"`
+	Order      string   `schema:"order,omitempty"`
+	OrderBy    string   `schema:"order_by,omitempty"`
+	PageNumber int      `schema:"page_number,omitempty"`
+	PageSize   int      `schema:"page_size,omitempty"`
+	Statuses   []string `schema:"statuses,omitempty"`
+}
+
+func (a *AnchorPlatformAPIService) getAnchorTransactions(ctx context.Context, skipAuthentication bool, queryParams GetTransactionsQueryParams) (*http.Response, error) {
+	// Path
+	u, err := url.Parse(a.AnchorPlatformBasePlatformURL)
+	if err != nil {
+		return nil, fmt.Errorf("creating url to GET anchor transactions: %w", err)
+	}
+	u = u.JoinPath("transactions")
+
+	// Query parameters
+	queryParamsEncoder := schema.NewEncoder()
+	params := url.Values{}
+	err = queryParamsEncoder.Encode(queryParams, params)
+	if err != nil {
+		return nil, fmt.Errorf("encoding query params in getAnchorTransactions: %w", err)
+	}
+	u.RawQuery = params.Encode()
+
+	// request
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating request to GET anchor transactions: %w", err)
+	}
+	request.Header.Set("Content-Type", "application/json")
+
+	// (skippable) JWT token
+	if !skipAuthentication {
+		var token string
+		token, err = a.GetJWTToken(nil)
+		if err != nil {
+			return nil, fmt.Errorf("getting jwt token in getAnchorTransactions: %w", err)
+		}
+		request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	}
+
+	// Do request
+	response, err := a.HttpClient.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("making getAnchorTransactions request to anchor platform: %w", err)
+	}
+
+	return response, nil
+}
+
+func (a *AnchorPlatformAPIService) IsAnchorProtectedByAuth(ctx context.Context) (bool, error) {
+	queryParams := GetTransactionsQueryParams{SEP: "24"}
+	resp, err := a.getAnchorTransactions(ctx, true, queryParams)
```

Comment:
Given that `getAnchorTransactions` is returning an `*http.Response`, I believe the callers should ensure that `response.Body` is closed to prevent a resource leak. 
```
defer resp.Body.Close()
```

Reply:
Good catch, thank you!

Reply:
Addressed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/10/commits/0a380e30939dd57210931c2405f26a8d09ff5986.